conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -83,6 +83,13 @@ public static <T> void addAll(Collection<T> addTo, Iterable<? extends T> element
 
     @SuppressWarnings("unchecked")
     public static <T> void addAllAndCheckNonNull(Collection<T> addTo, Iterable<? extends T> elementsToAdd) {
+        if (elementsToAdd instanceof ConjureIntegerList
+                || elementsToAdd instanceof ConjureDoubleList
+                || elementsToAdd instanceof ConjureSafeLongList) {
+            // Primitive lists will never have null elements.
+            addAll(addTo, elementsToAdd);
```

Comment:
We're skipping [this logic](https://github.com/palantir/safe-logging/blob/a055c3e3ced0b243ac226f5dfa160804eb6a0401/preconditions/src/main/java/com/palantir/logsafe/Preconditions.java#L320) for each element in the list. I wonder if this is actually a significant source of latency here? It'd be cool to have a jmh benchmark to validate that this gives us a performance gain we're expecting.

Reply:
That logic isn't the slow part -- the slow part is the adapting `long` to `SafeLong` since it doesn't appear to get JIT'd to a no-op like boxing.

Reply:
inlining some of the code here you get this:
```java
for (int i = 0; i < longArrayList.size(); i++) {
  long raw = longArrayList.get(i);
  SafeLong adapted = SafeLong.of(raw); /* It doesn't appear that the JIT erases this allocation */
  Preconditions.checkNotNull(adapted);
}
```
or this:
```java
for (int i = 0; i < intArrayList.size(); i++) {
  int raw = intArrayList.get(i);
  Long adapted = raw; /* It appears like the JIT erases this allocation */
  Preconditions.checkNotNull(adapted);
}
```